### PR TITLE
ARROW-14931: [Python] csv/orc format strings missing from some dataset docs

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

**Comment by westonpace** [comment]
@jorisvandenbossche

File: python/pyarrow/dataset.py [hunk 1]
```diff
@@ -757,10 +757,10 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         "part-{i}." + format.default_extname
     format : FileFormat or str
         The format in which to write the dataset. Currently supported:
-        "parquet", "ipc"/"feather". If a FileSystemDataset is being written
-        and `format` is not specified, it defaults to the same format as the
-        specified FileSystemDataset. When writing a Table or RecordBatch, this
-        keyword is required.
+        "parquet", "ipc"/"feather", and "csv". If a FileSystemDataset is
```
Comment (jorisvandenbossche) [inline python/pyarrow/dataset.py:760]:
```suggestion
        "parquet", "ipc"/"arrow"/"feather", and "csv". If a FileSystemDataset is
```

**Comment by jorisvandenbossche** [comment]
Good catch!

File: python/pyarrow/dataset.py [hunk 2]
```diff
@@ -757,10 +757,10 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         "part-{i}." + format.default_extname
     format : FileFormat or str
         The format in which to write the dataset. Currently supported:
-        "parquet", "ipc"/"feather". If a FileSystemDataset is being written
-        and `format` is not specified, it defaults to the same format as the
-        specified FileSystemDataset. When writing a Table or RecordBatch, this
-        keyword is required.
+        "parquet", "ipc"/"arrow"/"feather", and "csv". If a FileSystemDataset is
```
Comment (jorisvandenbossche) [inline python/pyarrow/dataset.py:760]: This line now has become one character too long ..

**Pull request closed without merging**